README.md

```diff
@@ -721,6 +726,39 @@ Discussions page][unicode].
 [BMP]: https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane
 [unicode]: https://github.com/peggyjs/peggy/discussions/15
 
+## Plugins API
+
+A plugin is an object with the `use(config, options)` method. That method will be
+called for all plugins in the `options.plugins` array, supplied to the `generate()`
+method.
+
+`use` accepts those parameters:
+- `config` — object with the following properties:
+  - `parser` — `Parser` object, by default the `peggy.parser` instance. That object
+    will be used to parse the grammar. Plugin can replace this object
+  - `passes` — mapping `{ [stage: string]: Pass[] }` that represents compilation
+    stages that would applied to the AST, returned by the `parser` object. That
+    mapping will contain at least the following keys:
+    - `check` — passes that check AST for correctness. They shouldn't change the AST
+    - `transform` — passes that performs various optimizations. They can change
+      the AST, add or remove nodes or their properties
+    - `generate` — passes used for actual code generating
+
+    Plugin that implement a pass usually should push it to the end of one of that
```

Comment:
"A Plugin that implements a pass usually should push it to the end of one of those arrays"